src/components/LanguagePicker/index.tsx

```diff
@@ -0,0 +1,226 @@
+import {
+  Box,
+  Flex,
+  FormControl,
+  FormLabel,
+  Input,
+  InputGroup,
+  InputRightElement,
+  Kbd,
+  Menu,
+  MenuList,
+  type MenuListProps,
+  type MenuProps,
+  Text,
+  type UseDisclosureReturn,
+  useEventListener,
+} from "@chakra-ui/react"
+
+import { Button } from "@/components/Buttons"
+import { BaseLink } from "@/components/Link"
+
+import MenuItem from "./MenuItem"
+import NoResultsCallout from "./NoResultsCallout"
+import { useLanguagePicker } from "./useLanguagePicker"
+
+type LanguagePickerProps = Omit<MenuListProps, "children"> & {
+  children: React.ReactNode
+  placement?: MenuProps["placement"]
+  handleClose?: () => void
+  menuState?: UseDisclosureReturn
+}
+
+const LanguagePicker = ({
+  children,
+  placement,
+  handleClose,
+  menuState,
+  ...props
+}: LanguagePickerProps) => {
+  const { t, refs, disclosure, filterValue, setFilterValue, filteredNames } =
+    useLanguagePicker(handleClose, menuState)
+  const { inputRef, firstItemRef, noResultsRef, footerRef } = refs
+  const { onClose } = disclosure
+
+  /**
+   * Adds a keydown event listener to focus filter input (\).
+   * @param {string} event - The keydown event.
+   */
+  useEventListener("keydown", (e) => {
+    if (e.key !== "\\") return
+    e.preventDefault()
+    inputRef.current?.focus()
```

Comment:
Couldn't we simplify this autofocus when the menu is opened, not just when the `keydown` is pressed? I say this because we should also autofocus the input when the user clicks the button.

There is a pretty standard user behavior where you click the button and then just start typing. You can't do that right now.

I think this was the original idea and I saw it working before. Just calling it out.